docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,457 @@
+ * @param[out] endorsements_buffer The buffer containing the endorsements to parse.
+ * @param[out] endorsements_buffer_size The size of the **endorsements_buffer**.
+ *
+ * @retval OE_OK The evidence and endorsements were successfully created.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval OE_OUT_OF_MEMORY Failed to allocate memory.
+ *
+ */
+oe_result_t oe_get_evidence(
+    uint32_t flags,
+    const uint8_t* custom_claims,
+    size_t custom_claims_size,
+    const void* opt_params,
+    size_t opt_params_size,
+    uint8_t** evidence_buffer,
+    size_t* evidence_buffer_size,
+    uint8_t** endorsements_buffer,
+    size_t* endorsements_buffer_size);
+
+/**
+ * Free up any resources allocated by oe_get_evidence()
+ *
+ * @param[in] evidence_buffer THe buffer containing the evidence.
+ * @param[in] endorsements_buffer The buffer containing the endorsements.
+ */
+void oe_free_evidence(
+    uint8_t* endorsements_buffer,
+    uint8_t* evidence_buffer);
+
+/**
+ * Verify the integrity of the evidence and its signature,
+ * with endorsements that are associated with the evidence.
+ * This function works for both local and remote attestation.
+ *
+ * This function is available in the enclave as well as in the host.
+ *
+ * @param[in] evidence_buffer The buffer containing the evidence to verify.
+ * @param[in] evidence_buffer_size The size of the **evidence** buffer.
+ * @param[in] endorsements Optional The endorsement data that is associated with
+ * the evidence.
+ * @param[in] endorsements_size The size of the **endorsements** buffer.
+ * @param[in] input_validation_time Optional datetime to use when verifying
+ * evidence. If not specified, it will use the creation_datetime of the
+ * endorsements (if any endorsements are provided).
+ * @param[out] claims The list of claims.
+ * @param[out] claims_size The size of claims.
+ *
+ * @retval OE_OK The verification was successful.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ *
+ */
+oe_result_t oe_verify_evidence(
+    const uint8_t* evidence_buffer,
+    size_t evidence_buffer_size,
+    const uint8_t* endorsements,
+    size_t endorsements_size,
+    oe_datetime_t* input_validation_time,
+    uint8_t** claims,
+    size_t* claims_size);
+```
+
+### Claims
+As part of the claims form `oe_verify_evidence()`, there will be a validity datetime range that applies to the evidence and endorsements.  For more information see the design document `CustomAttestation.md`.
+
+### OE Host Verify Library
+The OE Host Verify library is a standalone library used for verifying remote reports outside
+the TEE/enclave. The function `oe_verfiy_remote_report()` will be deprecated and should use the
+upcoming plug-in mode to do verification.
+
+
+Alternates
+----------
+
+### Endorsement structure:
+- Multiple serializable structures were considered.  At the end the final design
+made sense given that it is generic and support serialization.  We considered
```

Comment:
Who's "we"? The point of this design process is to have the design for public scrutiny. This statement provides no information on the actual alternatives that were considered, nor the reasons for their rejection. A proposal is not required to present alternatives and if it's not going to actually provide actionable information, I would much rather this section be dropped as it reads like an attempt to forestall discussion of design alternatives.

Reply:
Removed.